packages/core/admin/server/services/role.js

```diff
@@ -29,6 +29,34 @@ const pickComparableFields = pick(COMPARABLE_FIELDS);
 
 const jsonClean = (data) => JSON.parse(JSON.stringify(data));
 
+const ROLE_EVENTS = {
+  CREATE: 'role.create',
+  UPDATE: 'role.update',
+  DELETE: 'role.delete',
+  PERMISSIONS: 'role.permissions',
+};
+
+/**
+ * Emit an event on role actions
+ * @param {string} action
+ * @param {Function} fn controller
+ * @returns {Function}
+ */
+const wrapWithEmitEvent =
```

Comment:
I'm not sure this abstraction is needed here. I see it's done like this in the entity manager, but in my opinion it's a little different because over there they do the [same bit of logic](https://github.com/strapi/strapi/blob/bf2162df83c03699e2c0b6c1db4ec37dfe7f5352/packages/core/content-manager/server/services/entity-manager.js#L18-L23) for several events

Whereas here I think the only thing common to all events is calling eventHub.emit, so I think we can remove the wrap function and call the event hub directly. It also removes the need for the `event === ROLE_EVENTS.PERMISSIONS` condition